viewer/core/src/public/migration/Cognite3DViewer.ts

```diff
@@ -1572,34 +1572,38 @@ export class Cognite3DViewer {
     if (this.isDisposed) {
       return;
     }
-    this.latestRequestId = requestAnimationFrame(this._boundAnimate);
 
-    const { display, visibility } = window.getComputedStyle(this.canvas);
-    const isVisible = visibility === 'visible' && display !== 'none';
-
-    if (isVisible) {
-      const { renderController } = this;
-      TWEEN.update(time);
-      const didResize = this.resizeIfNecessary();
-      if (didResize) {
-        this.requestRedraw();
-      }
-      this.controls.update(this.clock.getDelta());
-      renderController.update();
-      this.revealManager.update(this.camera);
-
-      if (renderController.needsRedraw || this.revealManager.needsRedraw || this._clippingNeedsUpdate) {
-        const frameNumber = this.renderer.info.render.frame;
-        const start = Date.now();
-        this.updateCameraNearAndFar(this.camera);
-        this.revealManager.render(this.camera);
-        renderController.clearNeedsRedraw();
-        this.revealManager.resetRedraw();
-        this._clippingNeedsUpdate = false;
-        const renderTime = Date.now() - start;
-
-        this._events.sceneRendered.fire({ frameNumber, renderTime, renderer: this.renderer, camera: this.camera });
+    try {
+      this.latestRequestId = -1;
+      const { display, visibility } = window.getComputedStyle(this.canvas);
+      const isVisible = visibility === 'visible' && display !== 'none';
+
+      if (isVisible) {
+        const { renderController } = this;
+        TWEEN.update(time);
+        const didResize = this.resizeIfNecessary();
+        if (didResize) {
+          this.requestRedraw();
+        }
+        this.controls.update(this.clock.getDelta());
+        renderController.update();
+        this.revealManager.update(this.camera);
+
+        if (renderController.needsRedraw || this.revealManager.needsRedraw || this._clippingNeedsUpdate) {
+          const frameNumber = this.renderer.info.render.frame;
+          const start = Date.now();
+          this.updateCameraNearAndFar(this.camera);
+          this.revealManager.render(this.camera);
+          renderController.clearNeedsRedraw();
+          this.revealManager.resetRedraw();
+          this._clippingNeedsUpdate = false;
+          const renderTime = Date.now() - start;
+
+          this._events.sceneRendered.fire({ frameNumber, renderTime, renderer: this.renderer, camera: this.camera });
+        }
       }
+    } finally {
+      this.latestRequestId = requestAnimationFrame(this._boundAnimate);
```

Comment:
What is the significance of moving this here? What is the purpose of `this.latestRequestId`?

Reply:
Oh, that should not be part of this PR - fixing.